python/ray/data/datasource/tensorflow_datasource.py

```diff
@@ -0,0 +1,67 @@
+import logging
+from typing import TYPE_CHECKING, Callable, Iterator, List
+
+from ray.data.block import Block, BlockMetadata, T
+from ray.data.datasource import Datasource, ReadTask
+
+if TYPE_CHECKING:
+    import tensorflow as tf
+
+logger = logging.getLogger(__name__)
+
+
+class SimpleTensorFlowDatasource(Datasource[T]):
+    """A datasource that let's you use TensorFlow datasets with Ray Data.
+
+    .. warning::
+        ``SimpleTensorFlowDataset`` doesn't support parallel reads. You should only use
+        this datasource for small datasets like MNIST or CIFAR.
+
+    Example:
+        >>> def dataset_factory():
+        ...     return tfds.load("cifar10", split=["train"], as_supervised=True)[0]
+        ...
+        >>> dataset = ray.data.read_datasource(
+        ...     SimpleTensorFlowDatasource(),
+        ...     parallelism=1,
+        ...     dataset_factory=dataset_factory
+        ... )
+        >>> features, label = dataset.take(1)[0]
+        >>> features.shape
+        TensorShape([32, 32, 3])
+        >>> label
+        <tf.Tensor: shape=(), dtype=int64, numpy=7>
```

Comment:
Could use suggestions on how to simplify this part of the example.

Reply:
The example LGTM, what bits of it are you wanting to simplify?

Reply:
Just thought the example felt a bit long. If it looks good to you, it's probably fine